worker/src/RTC/TransportCongestionControlServer.cpp

```diff
@@ -109,6 +112,162 @@ namespace RTC
+
+			return;
+		}
+
+		if (
+		  this->periodicWindowStartSeq &&
+		  this->packetArrivalTimes.lower_bound(*this->periodicWindowStartSeq) ==
+		    this->packetArrivalTimes.end())
+		{
+			// Start new feedback packet, cull old packets.
+			for (auto it = this->packetArrivalTimes.begin(); it != this->packetArrivalTimes.end() &&
+			                                                 it->first < seq &&
+			                                                 arrival_time - it->second >= 500;)
+			{
+				it = this->packetArrivalTimes.erase(it);
+			}
+		}
+
+		if (!this->periodicWindowStartSeq || seq < *this->periodicWindowStartSeq)
+		{
+			this->periodicWindowStartSeq = seq;
+		}
+
+		// We are only interested in the first time a packet is received.
+		if (this->packetArrivalTimes.find(seq) != this->packetArrivalTimes.end())
+		{
+			return;
+		}
+
+		this->packetArrivalTimes[seq] = arrival_time;
+		// Limit the range of sequence numbers to send feedback for.
+		auto first_arrival_time_to_keep = this->packetArrivalTimes.lower_bound(
+		  this->packetArrivalTimes.rbegin()->first - kMaxNumberOfPackets);
+
+		if (first_arrival_time_to_keep != this->packetArrivalTimes.begin())
+		{
+			this->packetArrivalTimes.erase(this->packetArrivalTimes.begin(), first_arrival_time_to_keep);
+
+			// |this->packetArrivalTimes| cannot be empty since we just added one element
+			// and the last element is not deleted.
+			// RTC_DCHECK(!this->packetArrivalTimes.empty());
+			this->periodicWindowStartSeq = this->packetArrivalTimes.begin()->first;
+		}
+	}
+
+	void TransportCongestionControlServer::SendPeriodicFeedbacks()
+	{
+		// |this->periodicWindowStartSeq| is the first sequence number to include in the
+		// current feedback packet. Some older may still be in the map, in case a
+		// reordering happens and we need to retransmit them.
+		if (!this->periodicWindowStartSeq)
+		{
+			return;
+		}
+
+		for (auto begin_iterator = this->packetArrivalTimes.lower_bound(*this->periodicWindowStartSeq);
+		     begin_iterator != this->packetArrivalTimes.cend();
+		     begin_iterator = this->packetArrivalTimes.lower_bound(*this->periodicWindowStartSeq))
+		{
+			int64_t next_sequence_number = BuildFeedbackPacket(
+			  *this->periodicWindowStartSeq, begin_iterator, this->packetArrivalTimes.cend());
+
+			// If build feedback packet fail, it will not be sent.
+			if (next_sequence_number < 0)
+			{
+				// Reset and create a new feedback packet for next periodic.
+				this->transportCcFeedbackPacket.reset(new RTC::RTCP::FeedbackRtpTransportPacket(
+				  this->transportCcFeedbackSenderSsrc, this->transportCcFeedbackMediaSsrc));
+
+				return;
+			}
+
+			this->periodicWindowStartSeq = next_sequence_number;
+			SendTransportCcFeedback();
+			// Note: Don't erase items from this->packetArrivalTimes after sending, in case
+			// they need to be re-sent after a reordering. Removal will be handled
+			// by OnPacketArrival once packets are too old.
```

Comment:
```suggestion
			// Note: Don't erase items from this->packetArrivalTimes after sending, in case
			// they need to be re-sent after a reordering. Removal will be handled by OnPacketArrival
			// once packets are too old.
```